src/main/typescript/data/common.ts

```diff
@@ -8,6 +9,10 @@ export interface CompositeNumberSource {
   source: number;
 }
 
+export const CompositeNumberSchema = z
+  .object({ source: z.number().default(0) })
+  .default({});
```

Comment:
I think this default should be `{ source: 0 }`. I'm not sure if we need the inner ones, but might be better.

Reply:
Wouldn't that be equivalent?

Reply:
I don't know if zod recursively adds defaults. If yes then it is. If not, then no. You'd just have to try it.